### PR TITLE
Amazon Bedrock Integration

### PR DESCRIPTION
**Pull request by MinuraPunchihewa**
## Description

This PR adds the ML integration for Amazon Bedrock.

Fixes https://github.com/mindsdb/mindsdb_private/issues/611

## Type of change

- [X] ⚡ New feature (non-breaking change which adds functionality)

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [X] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/bedrock_handler/bedrock_handler.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+        Raises:
+            Exception: If the input does not match the configuration of the model.
+
+        Returns:
+            pd.DataFrame: Input data with the predicted values in a new column.
+        """
+        args = self.model_storage.json_get('args')
+        handler_model_params = args['handler_model_params']
+        mode = args['handler_model_params']['mode']
```
[the remark this answers is not in the document]
Comment (MinuraPunchihewa) [reply mindsdb/integrations/handlers/bedrock_handler/bedrock_handler.py:86]: > Cleaner: `mode = handler_model_params['mode']`

Done.

> Also, looking at the data model this is optional, so perhaps it'd be better to do `.get('mode')`?

In the data model, if a mode is not provided, the default mode is set, therefore, at the predict stage, it should not be empty. Do you think we need to add another check here?

File: mindsdb/integrations/handlers/bedrock_handler/bedrock_handler.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        model_id = handler_model_params['model_id']
+        inference_config = handler_model_params.get('inference_config')
+        target = args['target']
+
+        # Run predict for the default mode.
+        if mode == 'default':
+            prompts = self._prepare_data_for_default_mode(df, args)
+            predictions = self._predict_for_default_mode(model_id, prompts, inference_config)
+
```
[the remark this answers is not in the document]
Comment (MinuraPunchihewa) [reply mindsdb/integrations/handlers/bedrock_handler/bedrock_handler.py:95]: The data model also validates if a supported mode is provided via the `check_if_mode_is_supported` validator. Should we add another check here?

File: mindsdb/integrations/handlers/bedrock_handler/bedrock_handler.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        """
+        predictions = []
+        bedrock_runtime_client = create_amazon_bedrock_runtime_client(
+            **self.engine_storage.get_connection_args()
+        )
+
+        for prompt in prompts:
+            response = bedrock_runtime_client.converse(
+                modelId=model_id,
+                messages=[prompt],
```
[the remark this answers is not in the document]
Comment (MinuraPunchihewa) [reply mindsdb/integrations/handlers/bedrock_handler/bedrock_handler.py:166]: Understood. I was hoping to support a couple of other modes, Patricio. I was thinking that now that you've reviewed it and the overall structure looks fine, I can add support for `conversational`, `conversational-full` and `embeddings`.
I will definitely explain this in the docs.

**Comment by MinuraPunchihewa** [comment]
Hey @paxcema,
As we discussed, I have added support for `conversational` mode with the `prompt_template` and `question_column` (with an optional `context_column`) parameters until we decide on the next steps.
I have also added the documentation.
I have also refactored the code a little bit in adding support for this mode. I'd appreciate another review if it is not too much trouble.

**Comment by MinuraPunchihewa** [comment]
Thank you, @paxcema! @ZoranPandovski Let me know if you think there are any other changes required.

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**